pkgs/applications/audio/shortwave/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv, fetchFromGitLab, cargo, dbus, desktop-file-utils, gdk-pixbuf,
+  gettext, glib, gst_all_1, gtk3, libhandy, meson, ninja, openssl,
+  pkgconfig, python3, rust, rustc, rustPlatform, sqlite, wrapGAppsHook
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "shortwave";
+  version = "1.0.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "Shortwave";
+    rev = version;
+    sha256 = "13lhlh75vw02vkcknl4nvy0yvpdf0qx811mmyja8bzs4rj1j9kr8";
+  };
+
+  cargoSha256 = "0aph5z54a6i5p8ga5ghhx1c9hjc8zdw5pkv9inmanca0bq3hkdlh";
+
+  nativeBuildInputs = [
+    cargo desktop-file-utils gettext glib meson ninja pkgconfig
+    python3 rustc wrapGAppsHook
+  ];
+
+  buildInputs = [
+    dbus gdk-pixbuf gtk3 libhandy openssl sqlite
```

Comment:
It works because gtk3 propagates `glib` but `glib` should be here as well:

https://gitlab.gnome.org/World/Shortwave/-/blob/c3b47d7fcd5b2f0f1324f6668c58db843cc8e6bb/meson.build#L12-13